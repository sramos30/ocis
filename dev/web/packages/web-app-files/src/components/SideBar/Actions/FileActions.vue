<template>
  <oc-list id="oc-files-actions-sidebar" class="oc-mt-s">
    <action-menu-item
      v-for="(action, index) in actions"
      :key="`action-${index}`"
      :action="action"
      :items="resources"
      class="oc-py-xs"
    />
  </oc-list>
</template>

<script lang="ts">
import { mapGetters } from 'vuex'
import ActionMenuItem from '../../ActionMenuItem.vue'

import FileActions from '../../../mixins/fileActions'
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  name: 'FileActions',
  components: { ActionMenuItem },
  mixins: [FileActions],
  computed: {
    ...mapGetters('Files', ['highlightedFile']),

    resources() {
      return [this.highlightedFile]
    },

    actions() {
      return this.$_fileActions_getAllAvailableActions(this.resources)
    }
  }
})
</script>

<style lang="scss">
#oc-files-actions-sidebar {
  > li a,
  > li a:hover {
    color: var(--oc-color-swatch-passive-default);
    display: inline-flex;
    gap: 10px;
    vertical-align: top;
  }
}
</style>
