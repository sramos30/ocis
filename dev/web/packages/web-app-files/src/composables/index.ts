export * from './fileListHeaderPosition'
export * from './pagination'
export * from './resourcesViewDefaults'
export * from './router'
export * from './sort'

declare module 'vue/types/vue' {
  interface Vue {
    [key: string]: any
  }
}
