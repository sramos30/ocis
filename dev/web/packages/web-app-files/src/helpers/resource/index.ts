export * from './asset'
export * from './filter'
export * from './privatePreviewBlob'
export * from './publicPreviewUrl'
export * from './resource'
export * from './sameResource'
