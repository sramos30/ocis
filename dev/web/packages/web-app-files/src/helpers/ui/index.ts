export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
