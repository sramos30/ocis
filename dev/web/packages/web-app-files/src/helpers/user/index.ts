export * from './avatarUrl'
export * from './types'
