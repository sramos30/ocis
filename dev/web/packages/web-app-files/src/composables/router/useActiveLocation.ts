import { ref, Ref, watch } from '@vue/composition-api'
import { useRoute, useRouter } from 'web-pkg/src/composables'
import { ActiveRouteDirectorFunc } from '../../router'

/**
 * watches the current route and re-evaluates the provided active location director
 * on each route name update.
 *
 * @param director
 * @param comparatives
 */
export const useActiveLocation = <T extends string>(
  director: ActiveRouteDirectorFunc<T>,
  ...comparatives: T[]
): Ref<boolean> => {
  const value = ref(false)
  const router = useRouter()
  const currentRoute = useRoute()
  watch(
    currentRoute,
    () => {
      value.value = director(router, ...comparatives)
    },
    { immediate: true }
  )
  return value
}
