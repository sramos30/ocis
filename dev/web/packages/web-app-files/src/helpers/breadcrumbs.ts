import { bus } from 'web-pkg/src/instance'

export interface BreadcrumbItem {
  text: string
  to?: string
  allowContextActions?: boolean
  onClick?: () => void
}

export const breadcrumbsFromPath = (
  currentPath: string,
  resourcePath: string
): BreadcrumbItem[] => {
  const pathSplit = (p = '') => p.split('/').filter(Boolean)
  const current = pathSplit(currentPath)
  const resource = pathSplit(resourcePath)

  return resource.map(
    (text, i) =>
      ({
        allowContextActions: true,
        text,
        to: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/')
      } as BreadcrumbItem)
  )
}

export const concatBreadcrumbs = (...items: BreadcrumbItem[]): BreadcrumbItem[] => {
  const last = items.pop()

  return [
    ...items,
    {
      allowContextActions: last.allowContextActions,
      text: last.text,
      onClick: () => bus.publish('app.files.list.load')
    }
  ]
}
