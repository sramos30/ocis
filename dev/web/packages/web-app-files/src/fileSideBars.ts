import FileDetails from './components/SideBar/Details/FileDetails.vue'
import FileDetailsMultiple from './components/SideBar/Details/FileDetailsMultiple.vue'
import FileActions from './components/SideBar/Actions/FileActions.vue'
import FileVersions from './components/SideBar/Versions/FileVersions.vue'
import FileShares from './components/SideBar/Shares/FileShares.vue'
import FileLinks from './components/SideBar/Links/FileLinks.vue'
import NoSelection from './components/SideBar/NoSelection.vue'
import SpaceActions from './components/SideBar/Actions/SpaceActions.vue'
import SpaceDetails from './components/SideBar/Details/SpaceDetails.vue'
import SpaceShares from './components/SideBar/Shares/SpaceShares.vue'
import { isLocationSpacesActive, isLocationTrashActive, isLocationPublicActive } from './router'
import { spaceRoleEditor, spaceRoleManager } from './helpers/share'
import { Panel } from '../../web-pkg/src/components/sidebar'

import { Resource } from './helpers/resource'
import { User } from './helpers/user'
import Router from 'vue-router'

function $gettext(msg: string): string {
  return msg
}

const panelGenerators: (({
  rootFolder,
  highlightedFile,
  router,
  multipleSelection,
  user,
  capabilities
}: {
  rootFolder: boolean
  highlightedFile: Resource
  router: Router
  multipleSelection: boolean
  user: User
  capabilities: any
}) => Panel)[] = [
  // We don't have file details in the trashbin, yet.
  // Only allow `actions` panel on trashbin route for now.
  ({ rootFolder, highlightedFile }): Panel => ({
    app: 'no-selection-item',
    icon: 'questionnaire-line',
    title: $gettext('Details'),
    component: NoSelection,
    default: () => true,
    get enabled() {
      return rootFolder && highlightedFile?.type !== 'space'
    }
  }),
  ({ router, multipleSelection, rootFolder }) => ({
    app: 'details-item',
    icon: 'questionnaire-line',
    title: $gettext('Details'),
    component: FileDetails,
    default:
      !isLocationTrashActive(router, 'files-trash-personal') &&
      !isLocationTrashActive(router, 'files-trash-spaces-project'),
    get enabled() {
      return (
        !isLocationTrashActive(router, 'files-trash-personal') &&
        !isLocationTrashActive(router, 'files-trash-spaces-project') &&
        !multipleSelection &&
        !rootFolder
      )
    }
  }),
  ({ multipleSelection, rootFolder }) => ({
    app: 'details-multiple-item',
    icon: 'questionnaire-line',
    title: $gettext('Details'),
    component: FileDetailsMultiple,
    default: () => true,
    get enabled() {
      return multipleSelection && !rootFolder
    }
  }),
  ({ router, highlightedFile }) => ({
    app: 'details-space-item',
    icon: 'questionnaire-line',
    title: $gettext('Details'),
    component: SpaceDetails,
    default: isLocationSpacesActive(router, 'files-spaces-projects'),
    get enabled() {
      return highlightedFile?.type === 'space'
    }
  }),
  ({ router, multipleSelection, rootFolder }) => ({
    app: 'actions-item',
    icon: 'slideshow-3',
    title: $gettext('Actions'),
    component: FileActions,
    default:
      isLocationTrashActive(router, 'files-trash-personal') ||
      isLocationTrashActive(router, 'files-trash-spaces-project'),
    get enabled() {
      return !multipleSelection && !rootFolder
    }
  }),
  ({ highlightedFile, user }) => ({
    app: 'space-actions-item',
    icon: 'slideshow-3',
    title: $gettext('Actions'),
    component: SpaceActions,
    get enabled() {
      if (highlightedFile?.type !== 'space') {
        return false
      }
      return [
        ...highlightedFile.spaceRoles[spaceRoleManager.name],
        ...highlightedFile.spaceRoles[spaceRoleEditor.name]
      ].includes(user.uuid)
    }
  }),
  ({ capabilities, router, multipleSelection, rootFolder }) => ({
    app: 'sharing-item',
    icon: 'group',
    title: $gettext('People'),
    component: FileShares,
    get enabled() {
      if (multipleSelection || rootFolder) return false
      if (
        isLocationTrashActive(router, 'files-trash-personal') ||
        isLocationTrashActive(router, 'files-trash-spaces-project') ||
        isLocationPublicActive(router, 'files-public-files')
      ) {
        return false
      }

      if (capabilities.files_sharing) {
        return capabilities.files_sharing.api_enabled
      }
      return false
    }
  }),
  ({ highlightedFile }) => ({
    app: 'space-share-item',
    icon: 'group',
    title: $gettext('Members'),
    component: SpaceShares,
    get enabled() {
      return highlightedFile?.type === 'space'
    }
  }),
  ({ capabilities, router, multipleSelection, rootFolder, highlightedFile }) => ({
    app: 'links-item',
    icon: 'link',
    title: $gettext('Links'),
    component: FileLinks,
    get enabled() {
      if (multipleSelection || (rootFolder && highlightedFile?.type !== 'space')) return false
      if (
        isLocationTrashActive(router, 'files-trash-personal') ||
        isLocationTrashActive(router, 'files-trash-spaces-project') ||
        isLocationPublicActive(router, 'files-public-files')
      ) {
        return false
      }

      if (capabilities.files_sharing) {
        return capabilities.files_sharing.public.enabled
      }
      return false
    }
  }),
  ({ capabilities, highlightedFile, router, multipleSelection, rootFolder }) => ({
    app: 'versions-item',
    icon: 'git-branch',
    title: $gettext('Versions'),
    component: FileVersions,
    get enabled() {
      if (multipleSelection || rootFolder) return false
      if (
        isLocationTrashActive(router, 'files-trash-personal') ||
        isLocationTrashActive(router, 'files-trash-spaces-project') ||
        isLocationPublicActive(router, 'files-public-files')
      ) {
        return false
      }
      return !!capabilities.core && highlightedFile && highlightedFile.type !== 'folder'
    }
  })
]

export default panelGenerators
