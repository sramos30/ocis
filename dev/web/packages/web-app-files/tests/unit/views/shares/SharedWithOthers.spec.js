import { mount } from '@vue/test-utils'
import { getStore, localVue } from '../views.setup.js'
import FileActions from '@files/src/mixins/fileActions.js'
import SharedWithOthers from '@files/src/views/shares/SharedWithOthers.vue'
import SharedData from '@/__fixtures__/sharedFiles.js'
import { createLocationShares } from '../../../../src/router'
import { buildSharedResource } from '../../../../src/helpers/resources'
import { Settings } from 'luxon'
const resourcesList = SharedData.json().ocs.data.map((resource) => buildSharedResource(resource))

const router = {
  push: jest.fn(),
  afterEach: jest.fn(),
  currentRoute: {
    ...createLocationShares('files-shares-with-others'),
    query: {}
  },
  resolve: (r) => {
    return { href: r.name }
  }
}

const stubs = {
  'app-bar': true,
  'app-loading-spinner': true,
  'no-content-message': true,
  'resource-table': true,
  'context-actions': true,
  pagination: true,
  'list-info': true,
  'oc-resource': true,
  'oc-checkbox': true,
  'oc-avatars': true,
  'oc-resource-size': true,
  'oc-button': true,
  'oc-drop': true
}

const appLoadingSpinnerStub = 'app-loading-spinner-stub'
const noContentStub = 'no-content-message-stub'
const filesTableStub = 'resource-table-stub'
const filesTableSelector = '#files-shared-with-others-table'
const contextActionsStub = 'context-actions-stub'
const listInfoStub = 'list-info-stub'
const paginationStub = 'pagination-stub'

describe('SharedWithOthers view', () => {
  beforeAll(() => {
    Settings.defaultZone = 'utc'
  })

  it('should show the app-loading-spinner component when the wrapper is still loading', () => {
    const wrapper = getMountedWrapper({ loading: true })

    expect(wrapper.find(appLoadingSpinnerStub).exists()).toBeTruthy()
    expect(wrapper.find(noContentStub).exists()).toBeFalsy()
    expect(wrapper.find(filesTableStub).exists()).toBeFalsy()
  })
  describe('when the wrapper is not loading anymore', () => {
    it('should show the no content message component if the paginated resources is empty', () => {
      const wrapper = getMountedWrapper()

      expect(wrapper.find(appLoadingSpinnerStub).exists()).toBeFalsy()
      expect(wrapper.find(filesTableStub).exists()).toBeFalsy()
      expect(wrapper.find(noContentStub).exists()).toBeTruthy()
    })
    describe('when length of paginated resources is greater than zero', () => {
      const wrapper = getMountedWrapper({ paginatedResources: resourcesList })

      it('should not show the no content message component', () => {
        expect(wrapper.find(noContentStub).exists()).toBeFalsy()
      })

      it('should load the resource table with correct props', () => {
        stubs['resource-table'] = false
        const wrapper = getMountedWrapper({ paginatedResources: resourcesList })
        const filesTable = wrapper.find(filesTableSelector)

        expect(filesTable.exists()).toBeTruthy()
        expect(filesTable).toMatchSnapshot()

        stubs['resource-table'] = true
      })

      describe('context menu', () => {
        let wrapper
        const selectedResources = [resourcesList[0], resourcesList[1]]
        const notSelectedResources = [resourcesList[2]]
        beforeEach(() => {
          stubs['resource-table'] = false

          wrapper = getMountedWrapper({
            selectedFiles: selectedResources,
            paginatedResources: resourcesList,
            paginationPage: 1,
            paginationPages: 2
          })
        })
        afterEach(() => {
          stubs['resource-table'] = true
        })
        it('should show the context actions for every selected resource', () => {
          selectedResources.forEach((selectedResource) => {
            const fileRow = wrapper.find(`[data-item-id="${selectedResource.id}"]`)
            const contextMenu = fileRow.find(contextActionsStub)
            expect(contextMenu.exists()).toBeTruthy()
            expect(contextMenu.props().items).toMatchObject(selectedResources)
          })
        })
        it('should not show the context actions for a resource that is not selected', () => {
          notSelectedResources.forEach((notSelectedResource) => {
            const fileRow = wrapper.find(`[data-item-id="${notSelectedResource.id}"]`)
            const contextMenu = fileRow.find(contextActionsStub)
            expect(contextMenu.exists()).toBeFalsy()
          })
        })
      })

      it('should show the list info component for the resource table', () => {
        stubs['resource-table'] = false

        const wrapper = getMountedWrapper({
          paginationPage: 1,
          paginationPages: 1,
          paginatedResources: resourcesList
        })
        const listInfo = wrapper.find(listInfoStub)

        expect(listInfo.exists()).toBeTruthy()
        expect(listInfo.props()).toMatchObject({
          files: resourcesList.length,
          folders: 0,
          size: null
        })
        stubs['resource-table'] = true
      })

      describe('pagination component', () => {
        it('should be visible if the "paginationPages" is greater than one', () => {
          stubs['resource-table'] = false
          const wrapper = getMountedWrapper({
            paginationPage: 1,
            paginationPages: 2,
            paginatedResources: resourcesList
          })

          const pagination = wrapper.find(paginationStub)

          expect(pagination.exists()).toBeTruthy()
          expect(pagination.props()).toMatchObject({
            pages: 2,
            currentPage: 1
          })
          stubs['resource-table'] = true
        })
      })

      it('should call "$_fileActions_triggerDefaultAction" method if "fileClick" event is emitted from the resource table component', () => {
        const spyFileActionsTriggerDefaultAction = jest
          .spyOn(FileActions.methods, '$_fileActions_triggerDefaultAction')
          .mockImplementation()
        const wrapper = getMountedWrapper({
          paginatedResources: resourcesList,
          paginationPage: 1,
          paginationPages: 2
        })
        const filesTable = wrapper.find(filesTableStub)
        expect(spyFileActionsTriggerDefaultAction).toHaveBeenCalledTimes(0)

        filesTable.vm.$emit('fileClick')

        expect(spyFileActionsTriggerDefaultAction).toHaveBeenCalledTimes(1)
      })

      it('should call "rowMounted" method if "rowMounted" event is emitted from the resource table component', () => {
        const spyRowMounted = jest
          .spyOn(SharedWithOthers.methods, 'rowMounted')
          .mockImplementation()
        const wrapper = getMountedWrapper({
          paginatedResources: resourcesList,
          paginationPage: 1,
          paginationPages: 2
        })
        const filesTable = wrapper.find(filesTableStub)
        expect(spyRowMounted).toHaveBeenCalledTimes(0)

        filesTable.vm.$emit('rowMounted')

        expect(spyRowMounted).toHaveBeenCalledTimes(1)
      })
    })
  })

  function getMountedWrapper({
    selectedFiles = [],
    loading = false,
    paginatedResources = [],
    paginationPages = 12,
    paginationPage = 21
  } = {}) {
    const store = getStore({
      selectedFiles,
      totalFilesCount: { files: paginatedResources.length, folders: 0 }
    })
    const component = {
      ...SharedWithOthers,
      created: jest.fn(),
      mounted: jest.fn(),
      setup: () => ({
        ...SharedWithOthers.setup(),
        loadResourcesTask: {
          isRunning: loading,
          perform: jest.fn()
        },
        paginatedResources: paginatedResources,
        paginationPages: paginationPages,
        paginationPage: paginationPage,
        handleSort: jest.fn()
      })
    }
    return mount(component, {
      localVue,
      store: store,
      stubs,
      mocks: {
        $route: router.currentRoute,
        $router: router
      }
    })
  }
})
