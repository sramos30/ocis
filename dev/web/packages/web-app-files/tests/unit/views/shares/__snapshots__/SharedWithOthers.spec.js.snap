// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedWithOthers view when the wrapper is not loading anymore when length of paginated resources is greater than zero should load the resource table with correct props 1`] = `
<table class="files-table oc-table oc-table-hover oc-table-sticky files-table-squashed" id="files-shared-with-others-table">
  <thead class="oc-thead">
    <tr tabindex="-1" class="oc-table-header-row">
      <th class="oc-th oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-text-nowrap oc-table-header-cell oc-table-header-cell-select oc-pl-m " style="top: 0px;"><span class="oc-table-thead-content"><div class="resource-table-select-all"><oc-checkbox-stub id="resource-table-select-all" label="Select all resources" hidelabel="true" size="large"></oc-checkbox-stub></div></span>
        <!---->
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-nowrap oc-table-header-cell oc-table-header-cell-name" aria-sort="none" style="top: 0px;"><span class="oc-table-thead-content header-text">Name</span>
        <oc-button-stub type="button" size="medium" arialabel="Sort by name" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-button-sort oc-invisible"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-sharedWith" aria-sort="none" style="top: 0px;"><span class="oc-table-thead-content header-text">Shared with</span>
        <oc-button-stub type="button" size="medium" arialabel="Sort by sharedWith" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-button-sort oc-invisible"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-sdate" aria-sort="none" style="top: 0px;"><span class="oc-table-thead-content header-text">Shared on</span>
        <oc-button-stub type="button" size="medium" arialabel="Sort by sdate" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-button-sort oc-invisible"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-actions oc-pr-m" style="top: 0px;"><span class="oc-table-thead-content header-text">Actions</span>
        <!---->
      </th>
    </tr>
  </thead>
  <tbody>
    <tr tabindex="-1" data-item-id="38" draggable="false" class="oc-tbody-tr oc-tbody-tr-38">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select oc-pl-m ">
        <oc-checkbox-stub id="resource-table-select-38" value="" option="[object Object]" label="Select folder" hidelabel="true" size="large"></oc-checkbox-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="resource-table-resource-wrapper">
          <oc-resource-stub folderlink="[object Object]" parentfolderlink="[object Object]" resource="[object Object]" parentfoldernamedefault="All files and folders" ispathdisplayed="true" isresourceclickable="true"></oc-resource-stub>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="resource-table-edit-name"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <oc-button-stub type="button" size="medium" submit="button" variation="inverse" appearance="raw" justifycontent="center" gapsize="medium">
          <oc-avatars-stub items="[object Object]" stacked="true" istooltipdisplayed="true" maxdisplayed="3" accessibledescription=" This folder is shared via 1 link" class="resource-table-people"></oc-avatars-stub>
        </oc-button-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate" aria-label="9 months ago (June 9, 2021, 12:44 PM UTC)"><span tabindex="0">9 months ago</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions oc-pr-m">
        <div class="resource-table-actions">
          <oc-button-stub type="button" size="medium" arialabel="Show context menu" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="context-menu-trigger-38" class="resource-table-btn-action-dropdown"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></oc-button-stub>
          <oc-drop-stub dropid="context-menu-drop-38" popperoptions="[object Object]" toggle="#context-menu-trigger-38" position="bottom-start" mode="click" closeonclick="true" paddingsize="small"></oc-drop-stub>
        </div>
      </td>
    </tr>
    <tr tabindex="-1" data-item-id="64" draggable="false" class="oc-tbody-tr oc-tbody-tr-64">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select oc-pl-m ">
        <oc-checkbox-stub id="resource-table-select-64" value="" option="[object Object]" label="Select folder" hidelabel="true" size="large"></oc-checkbox-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="resource-table-resource-wrapper">
          <oc-resource-stub folderlink="[object Object]" parentfolderlink="[object Object]" resource="[object Object]" parentfoldernamedefault="All files and folders" ispathdisplayed="true" isresourceclickable="true"></oc-resource-stub>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="resource-table-edit-name"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <oc-button-stub type="button" size="medium" submit="button" variation="inverse" appearance="raw" justifycontent="center" gapsize="medium">
          <oc-avatars-stub items="[object Object]" stacked="true" istooltipdisplayed="true" maxdisplayed="3" accessibledescription=" This folder is shared via 1 link" class="resource-table-people"></oc-avatars-stub>
        </oc-button-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate" aria-label="9 months ago (June 9, 2021, 12:44 PM UTC)"><span tabindex="0">9 months ago</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions oc-pr-m">
        <div class="resource-table-actions">
          <oc-button-stub type="button" size="medium" arialabel="Show context menu" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="context-menu-trigger-64" class="resource-table-btn-action-dropdown"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></oc-button-stub>
          <oc-drop-stub dropid="context-menu-drop-64" popperoptions="[object Object]" toggle="#context-menu-trigger-64" position="bottom-start" mode="click" closeonclick="true" paddingsize="small"></oc-drop-stub>
        </div>
      </td>
    </tr>
    <tr tabindex="-1" data-item-id="39" draggable="false" class="oc-tbody-tr oc-tbody-tr-39">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select oc-pl-m ">
        <oc-checkbox-stub id="resource-table-select-39" value="" option="[object Object]" label="Select folder" hidelabel="true" size="large"></oc-checkbox-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="resource-table-resource-wrapper">
          <oc-resource-stub folderlink="[object Object]" parentfolderlink="[object Object]" resource="[object Object]" parentfoldernamedefault="All files and folders" ispathdisplayed="true" isresourceclickable="true"></oc-resource-stub>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="resource-table-edit-name"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <oc-button-stub type="button" size="medium" submit="button" variation="inverse" appearance="raw" justifycontent="center" gapsize="medium">
          <oc-avatars-stub items="[object Object]" stacked="true" istooltipdisplayed="true" maxdisplayed="3" accessibledescription=" This folder is shared via 1 link" class="resource-table-people"></oc-avatars-stub>
        </oc-button-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate" aria-label="9 months ago (June 9, 2021, 12:44 PM UTC)"><span tabindex="0">9 months ago</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions oc-pr-m">
        <div class="resource-table-actions">
          <oc-button-stub type="button" size="medium" arialabel="Show context menu" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="context-menu-trigger-39" class="resource-table-btn-action-dropdown"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></oc-button-stub>
          <oc-drop-stub dropid="context-menu-drop-39" popperoptions="[object Object]" toggle="#context-menu-trigger-39" position="bottom-start" mode="click" closeonclick="true" paddingsize="small"></oc-drop-stub>
        </div>
      </td>
    </tr>
    <tr tabindex="-1" data-item-id="30" draggable="false" class="oc-tbody-tr oc-tbody-tr-30">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select oc-pl-m ">
        <oc-checkbox-stub id="resource-table-select-30" value="" option="[object Object]" label="Select file" hidelabel="true" size="large"></oc-checkbox-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="resource-table-resource-wrapper">
          <oc-resource-stub folderlink="[object Object]" parentfolderlink="[object Object]" resource="[object Object]" parentfoldernamedefault="All files and folders" ispathdisplayed="true" isresourceclickable="true"></oc-resource-stub>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="resource-table-edit-name"><span class="oc-icon oc-icon-s oc-icon-passive"><!----></span></oc-button-stub>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <oc-button-stub type="button" size="medium" submit="button" variation="inverse" appearance="raw" justifycontent="center" gapsize="medium">
          <oc-avatars-stub items="[object Object],[object Object]" stacked="true" istooltipdisplayed="true" maxdisplayed="3" accessibledescription="This file is shared via 1 invite This file is shared via 1 link" class="resource-table-people"></oc-avatars-stub>
        </oc-button-stub>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate" aria-label="9 months ago (June 9, 2021, 12:37 PM UTC)"><span tabindex="0">9 months ago</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions oc-pr-m">
        <div class="resource-table-actions">
          <oc-button-stub type="button" size="medium" arialabel="Show context menu" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="context-menu-trigger-30" class="resource-table-btn-action-dropdown"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></oc-button-stub>
          <oc-drop-stub dropid="context-menu-drop-30" popperoptions="[object Object]" toggle="#context-menu-trigger-30" position="bottom-start" mode="click" closeonclick="true" paddingsize="small"></oc-drop-stub>
        </div>
      </td>
    </tr>
  </tbody>
  <tfoot class="oc-table-footer">
    <tr class="oc-table-footer-row">
      <td colspan="5" class="oc-table-footer-cell">
        <pagination-stub pages="12" currentpage="21"></pagination-stub>
        <list-info-stub files="4" folders="0" class="oc-width-1-1 oc-my-s"></list-info-stub>
      </td>
    </tr>
  </tfoot>
</table>
`;
